### ISSUE TITLE
Watch freezes after some notifications

### ISSUE THREAD

**Issue by everypizza1** [InfiniTimeOrg/InfiniTime#2012]
### Verification

- [X] I searched for similar bug reports (including closed issues) and found none was relevant.

### What happened?

I received a notification and the touch screen froze for about 1 minute

### Reproduction steps

It doesn't consistently happen, but some things I've noticed:
* Only happens on notifications with emoji or with text in non-English languages.
* Only affects touchscreen, notification still scrolls.

### More details?

Only seems to affect notifications from YouTube for live streams and foreign language notifications from apps like holoplus.foregin

### Version

v1.14

### Companion app

Gadgetbridge 0.78.0

**Comment by minacode** [comment]
InfiniTime does not support all characters. Gadgetbridge might handle this somehow, but I don't know how for sure. 

1. Do you have an example of a notification that causes the bug? 
2. Is there a way to find out what text Gadgetbridge is actually sending to the watch?

**Comment by everypizza1** [comment]
One almost certain way that it's triggered is when someone I have subscribed to does a YouTube livestream. Gadgetbridge seems to be sending the notifications from YouTube with :red_circle: to replace the 🔴 in the notification. I don't have any notifications currently on the watch, but this is one from my notification history that caused it:
[Image]

(I did this through email so it might not work)


________________________________

**Comment by sbe-arg** [comment]
I'm noticing this as well. Specially with emoji notifications from whatsapp and signal.

Also using gadgetbridge.

**Comment by minacode** [comment]
Have you tried other companion apps?

**Comment by FintasticMan** [comment]
Could you try finding a reproducable case by trying it out in GadgetBridge's Debug menu? That allows you to send test notifications.

**Comment by sbe-arg** [comment]
> Could you try finding a reproducable case by trying it out in GadgetBridge's Debug menu? That allows you to send test notifications.

Been trying to reproduce it without any luck a message that might lag when coming from the app not necessarily lags when triggered via debug.

But ill keep checking

**Comment by everypizza1** [comment]
I also can't reproduce it in Gadgetbridge.


________________________________

**Comment by everypizza1** [comment]
I also can't reproduce this in Gadgetbridge.
________________________________

**Comment by everypizza1** [comment]
Just happened again with this notification:
![Screenshot_20240216-070543.jpg](https://github.com/InfiniTimeOrg/InfiniTime/assets/82726593/aa94084b-345b-408d-9666-b786be4456f2)

**Comment by minacode** [comment]
This was with Gadgedbridge? 
If yes, is there any chance of finding out what Gadgetbridge actually sends to the watch? 
It does transform the notification text to something InfiniTime should be able to show, right?

**Comment by everypizza1** [comment]
Gadgetbridge will always convert emojis to a description of them, like red_dot for 🔴, surrounded by colons. For example: `:red_dot:`.

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by minacode** [comment]
Ok, but can you somehow reproduce what it sent in the case of your faulty notification?

**Comment by everypizza1** [comment]
I've tried to reproduce many times with no success. It has just happened again though.
Watch:
![IMG_20240217_185644.jpg](https://github.com/InfiniTimeOrg/InfiniTime/assets/82726593/6b67a986-536b-4213-868e-3735337016a1)
Phone:
![Screenshot_20240217-185658.jpg](https://github.com/InfiniTimeOrg/InfiniTime/assets/82726593/76b79a69-5820-4459-b11d-bfa68bfeacc5)

**Comment by sbe-arg** [comment]
Do you have the heart rate monitor on? I noticed it only happens in my case when is on if of I don't have the freeze.

**Comment by minacode** [comment]
@everypizza1 
The notification contains a longer dash character, like "—".
In the title of your notification on the watch, the seems to be a strange character displayed right after "Pancakes" where the dash should be. I wonder what happened there.

@sbe-arg
Oh, no 😄
Let's see where this goes. Maybe there is another race somewhere. The heart rate has its own task if I remember correctly.

**Comment by everypizza1** [comment]
> @everypizza1 
> The notification contains a longer dash character, like "—".
> In the title of your notification on the watch, the seems to be a strange character displayed right after "Pancakes" where the dash should be. I wonder what happened there. …

I just realized that it skipped the long dash and went straight to the artist name. Apparently all Genius notifications have a long dash, I just tested with other songs. However others didn't cause it for some reason.

**Comment by Self-Perfection** [comment]
I probably have the same issue but with some variations:

In these hang state these function does not work:
* screen swipes (no reaction)
* single button press to go one screen back

But these function keep working:
* Seconds hand on analog watch face keeps ticking
* Longer button presses to get to clock or firmware version or reboot
* Double button press to switch to notifications screen

In my case waiting does not unfreeze watch. I have to reboot.

It have not been happening with previous firmware version, but have happened about 5-7 times with 1.14.0.

I have no idea what triggers freezing.

**Comment by FintasticMan** [comment]
Hmm, good observation. Maybe there's a race somewhere triggered by trying to go to sleep.

**Comment by timteichmann** [comment]
Not sure if any further confirmation is required. But I manage two daily driver PineTimes and have experienced regular freezes (every other day) that would require a reset with the setting "raise to wake" and "lower to sleep" enabled.

I can confirm that disabling "lower to sleep" as per @everypizza1 's comment (Cheers for posting that here) does resolve the issue. This seems to be the culprit.

**Comment by copyrights** [comment]
> I probably have the same issue but with some variations:
> 
> In these hang state these function does not work:
> 
>     * screen swipes (no reaction)
> 
>     * single button press to go one screen back
> 
> 
> But these function keep working:
> 
>     * Seconds hand on analog watch face keeps ticking
> 
>     * Longer button presses to get to clock or firmware version or reboot
> 
>     * Double button press to switch to notifications screen
> 
> 
> In my case waiting does not unfreeze watch. I have to reboot.
> 
> It have not been happening with previous firmware version, but have happened about 5-7 times with 1.14.0.
> 
> I have no idea what triggers freezing.

I have the same behavior. From what I have observed so far it happens like this
* A notification pops up
* the notification disappears regular after timeout, but display is still on.
* **Any touch interaction (or maybe other interaction too) triggers bug**
* If unnoticed battery drains fast
* Only reset (very long button press) helps

**Comment by everypizza1** [comment]
@copyrights Do you have lower-to-sleep turned on?

**Comment by copyrights** [comment]
Yes, Settings -> Wake Up -> Lower Wrist was on.  I disabled it now and will see if it mitigates this issue.

10.06.2024 22:52:59 Derry Tutt ***@***.***>:

> 
> @copyrights Do you have lower-to-sleep turned on?
>

**Comment by sbe-arg** [comment]
It hasn't happened to me in a while now no changes other then regular updates of gadgetbridge.

**Comment by mark9064** [comment]
Can anyone still repro this? Some race conditions around sleep/wake were resolved for 1.15
